internal/pkg/aws/partitions/partition.go

```diff
@@ -19,3 +18,17 @@ func (r Region) Partition() (endpoints.Partition, error) {
 	}
 	return partition, nil
 }
+
+// IsAvailableInRegion returns true if the service ID is available in the given region.
+func IsAvailableInRegion(sID string, region string) (bool, error) {
+	partition, err := Region(region).Partition()
+	if err != nil {
+		return false, err
+	}
+	regionServices, existInPartition := endpoints.RegionsForService(endpoints.DefaultPartitions(), partition.ID(), sID)
+	if !existInPartition {
+		return false, nil
+	}
+	_, existInRegion := regionServices[region]
```

Comment:
tiny tiny nit: after checking the doc I feel like these var names may better descriptions ⬇️ 
```suggestion
	regions, partitionOrServiceExists := endpoints.RegionsForService(endpoints.DefaultPartitions(), partition.ID(), sID)
	if !partitionOrServiceExists {
		return false, nil
	}
	_, existInRegion := regions[region]
```